apps/reference/docs/guides/auth/auth-mfa.mdx

```diff
@@ -0,0 +1,570 @@
+  ListFactors -->|1 or more factors| OpenAuth([User: Open authenticator]) --> Enter
+  ListFactors -->|0 factors| Setup[[Setup flow]]
+`} />
+
+These sets of APIs let you control the MFA experience that works for you. You
+can create flows where MFA is optional, mandatory for all or only specific
+groups of users.
+
+Once users have enrolled or signed-in with a factor, Supabase Auth adds
+additional metadata to the user's access token (JWT) that your application can
+use to allow or deny access.
+
+This information is represented by an [Authenticator Assurance
+Level](https://pages.nist.gov/800-63-3-Implementation-Resources/63B/AAL/), a
+standard measure about the assurance Supabase Auth has of the user's identity
+for that particular session. There are two levels recognized today:
+
+1. **Assurance Level 1: `aal1`**  
+Means that the user's identity was verified using a conventional login method
+such as email+password, magic link, one-time password, phone auth or social
+login.
+2. **Assurance Level 2: `aal2`**  
+Means that the user's identity was additionally verified using at least one
+second factor, such as a TOTP code.
+
+This assurance level is encoded in the `aal` claim in the JWT associated with
+the user. By decoding this value you can create custom authorization rules in
+your frontend, backend and database that will enforce the MFA policy that works
+for your application. JWTs without an `aal` claim are at the `aal1` level.
+
+## Adding to your app
+
+Adding multi-factor authentication to your app involves these three steps:
+
+1. **Add enrollment flow.**  
+You need to provide a UI within your app that your users will be able to set-up
+MFA in. You can add this right after sign-up, or as part of a separate flow in
+the settings portion of your app.
+2. **Add challenge step to login.**  
+If a user has set-up MFA, your app's login flow needs to present a challenge
+screen to the user asking them to prove they have access to the additional
+factor.
+3. **Enforce rules for MFA logins.**  
+Once your users have a way to enroll and log in with MFA, you need to enforce
+authorization rules across your app: on the frontend, backend, API servers or
+Row-Level Security policies.
+
+### Add enrollment flow
+
+Before users can secure their accounts with MFA, you need to provide them with
+a UI where they can set up their additional authentication factor. We call this
+the enrollment flow.
+
+Most applications add the enrollment flow in two places within their app:
+
+1. Right after login or sign up.  
+This lets users quickly set up MFA immediately after they log in or create an
+account. We recommend encouraging all users to set up MFA if that makes sense
+for your application. Many applications offer this as an opt-in step in an
+effort to reduce onboarding friction.
+2. From within a settings page.  
+Allows users to set up, disable or modify their MFA settings.
```

Comment:
I think we should leave out the recommendation and leave it up to the dev. We can likely assume they're already encouraging MFA for their users if they're going through this guide. This is just a suggestion though—will leave it up to you!
```suggestion
1. Right after login or sign up.  
This lets users set up MFA immediately after they log in or create an
account. Many applications offer this as an opt-in step in an
effort to reduce onboarding friction.
2. From within a settings page.  
Allows users to set up, disable, or modify their MFA settings.
```

Reply:
I actually would like to leave it as is. It lets users figure out exactly what they can do with MFA, and that there are some tricky choices to be made. It then also follows the recommendations with the RLS policies below...